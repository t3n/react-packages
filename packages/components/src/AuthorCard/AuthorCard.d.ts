/// <reference types="react" />
export interface AuthorValues {
    name: string;
    avatar: string;
}
export interface AuthorCardProps {
    title: string;
    articleType: string;
    author: AuthorValues;
    url: string;
}
declare const AuthorCard: ({ title, articleType, author, url }: AuthorCardProps) => JSX.Element;
export default AuthorCard;
