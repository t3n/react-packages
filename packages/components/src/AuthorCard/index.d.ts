import AuthorCard from './AuthorCard';
export * from './AuthorCard';
export default AuthorCard;
