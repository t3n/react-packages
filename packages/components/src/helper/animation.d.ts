import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
