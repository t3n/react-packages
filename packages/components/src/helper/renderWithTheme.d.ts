import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
