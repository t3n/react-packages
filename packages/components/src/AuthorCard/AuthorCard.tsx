import React from 'react';
import styled from 'styled-components';

import Badge from '../Badge';
import Card from '../Card';
import Heading from '../Heading';

export interface AuthorCardProps {
  title: string;
  articleType: string;
  author: {
    name: string;
    avatar: string;
  };
  url: string;
}

const Author = styled.div`
  display: flex;
  align-items: center;
`;

const AuthorCard = ({ title, articleType, author, url }: AuthorCardProps) => (
  <Card href={url} stretch>
    <div>
      <Badge variant="highlight" small>
        {articleType}
      </Badge>
    </div>

    <div style={{ flex: 1 }}>
      <Heading styleAs="h5" as="h2" mt={2} mb={8}>
        {title}
      </Heading>
    </div>

    <Author>
      <img
        src={author.avatar}
        width={40}
        style={{ marginRight: '1rem', borderRadius: '50%' }}
        alt={author.name}
      />
      {author.name}
    </Author>
  </Card>
);

export default AuthorCard;
