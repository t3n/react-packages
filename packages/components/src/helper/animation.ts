import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
