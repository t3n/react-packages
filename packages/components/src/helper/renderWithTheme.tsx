/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
