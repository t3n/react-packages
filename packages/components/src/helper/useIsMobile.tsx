import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
