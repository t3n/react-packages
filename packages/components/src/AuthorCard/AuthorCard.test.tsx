import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import AuthorCard from './AuthorCard';

const card = (
  <AuthorCard
    articleType="News"
    title="Auhor Card"
    url="https://t3n.de"
    author={{ name: 't3n Redaktion', avatar: '' }}
  />
);

test('Author Card matches snapshot', () => {
  const { container } = renderWithTheme(card, {});
  expect(container.firstChild).toMatchSnapshot();
});

test('Author Card renders content', () => {
  const { getByText } = renderWithTheme(card, {});

  expect(getByText(/t3n redaktion/i)).toBeDefined();
  expect(getByText(/news/i)).toBeDefined();
});
