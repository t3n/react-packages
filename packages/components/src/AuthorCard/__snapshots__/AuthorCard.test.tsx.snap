// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Card matches snapshot 1`] = `
.c0 {
  display: block;
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  -webkit-transition: box-shadow .15s ease-out,-webkit-transform .15s ease-out;
  -webkit-transition: box-shadow .15s ease-out,transform .15s ease-out;
  transition: box-shadow .15s ease-out,transform .15s ease-out;
  border: 1px solid #cccccc;
  border-radius: 4px;
  width: 100%;
  padding: 16px;
  color: #2a2a2a;
  box-shadow: 0 2px 10px rgba(0,0,0,.15);
  margin-bottom: 16px;
}

.c0:hover {
  box-shadow: 0 5px 20px 0 rgba(0,0,0,.2);
  -webkit-transform: translate3d(0,-2px,0);
  -ms-transform: translate3d(0,-2px,0);
  transform: translate3d(0,-2px,0);
}

.c0 .c5 {
  margin-left: -24px;
  margin-right: -24px;
  margin-top: -24px;
  margin-bottom: 24px;
}

.c0 .c5 .c4 {
  padding: 16px;
}

.c1 {
  font-weight: bold;
  background-color: #f9423a;
  padding: 0 2px;
  font-size: 0.875rem;
  border-radius: 0;
}

.c2 {
  font-family: "adelle-sans",sans-serif;
  font-size: 1.125rem;
  line-height: 1.5em;
  margin-top: 8px;
  margin-bottom: 56px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (min-width:30rem) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:30rem) {
  .c0 .c5 .c4 {
    padding: 24px;
  }
}

@media screen and (min-width:30rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media screen and (min-width:48rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media screen and (min-width:61.25rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

@media screen and (min-width:90rem) {
  .c2 {
    font-size: 1.25rem;
  }
}

@media screen and (min-width:30rem) {
  .c2 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:48rem) {
  .c2 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:61.25rem) {
  .c2 {
    line-height: 1.5em;
  }
}

@media screen and (min-width:90rem) {
  .c2 {
    line-height: 1.333em;
  }
}

<a
  class="c0"
  color="text.primary"
  href="https://t3n.de"
  width="1"
>
  <div>
    <span
      class="c1"
    >
      News
    </span>
  </div>
  <div
    style="flex: 1;"
  >
    <h2
      class="c2"
    >
      Auhor Card
    </h2>
  </div>
  <div
    class="c3"
  >
    <img
      alt="t3n Redaktion"
      src=""
      style="margin-right: 1rem; border-radius: 50%;"
      width="40"
    />
     
    t3n Redaktion
  </div>
</a>
`;
